Support for parameterization

Really like the clarity this plugin provides. I did notice that when running tests with parametrization, it does not show which parameter is being used.

for me, without the plugin, the output is without input parameters, 

```
========================================= test session starts =========================================
platform darwin -- Python 2.7.14, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/local/opt/python/bin/python2.7
cachedir: .cache
rootdir: /Users/GowthamSai, inifile:
plugins: testdox-1.0.1, spec-1.1.0, pspec-0.0.3, cov-2.4.0, celery-4.1.0
collected 3 items

demo.py::::Evaluating values PASSED                                                             [ 33%]
demo.py::::Evaluating values PASSED                                                             [ 66%]
demo.py::::Evaluating values FAILED                                                             [100%]

============================================== FAILURES ===============================================
__________________________________________ test_eval[6*9-42] __________________________________________

test_input = '6*9', expected = 42

    @pytest.mark.parametrize("test_input,expected", [
        ("3+5", 8),
        ("2+4", 6),
        ("6*9", 42),
    ])

    def test_eval(test_input, expected):
        "Evaluating values"
>       assert eval(test_input) == expected
E       AssertionError: assert 54 == 42
E        +  where 54 = eval('6*9')

demo.py:11: AssertionError
================================= 1 failed, 2 passed in 0.06 seconds ==================================
```

Can you paste the entire output, so that we can check OS, version and other details?

--- a/pytest_pspec/plugin.py
+++ b/pytest_pspec/plugin.py
@@ -36,6 +36,10 @@ def pytest_collection_modifyitems(config, items):
         parent = item.parent.obj
         node_parts = item.nodeid.split('::')
         node_str = node.__doc__ or node_parts[-1]
+
+        if hasattr(item, "callspec"):
+            node_str = node_str.format(**item.callspec.params)
+
         mode_str = node_parts[0]
         klas_str = ''
         node_parts_length = len(node_parts)

--- a/tests/test_plugin.py
+++ b/tests/test_plugin.py
@@ -2,6 +2,7 @@
 from __future__ import unicode_literals
 
 import pytest
+from _pytest.config import ExitCode
 
 
 class TestReport(object):
@@ -180,3 +181,39 @@ class TestReport(object):
 
         expected = 'This is PySpec Class'
         assert expected in result.stdout.str()
+
+    def test_parametrization_succeeds(self, testdir):
+        testdir.makepyfile("""
+            import pytest
+            @pytest.mark.parametrize("test_input, expected", [
+                ("3 + 5", 8),
+                ("2 + 4", 6),
+                ("6 * 9", 42),
+            ])
+            def test_math(test_input, expected):
+                '''Checking if {test_input} = {expected}...'''
+                assert eval(test_input) == expected
+        """)
+
+        result = testdir.runpytest('--pspec')
+
+        expected = "3 + 5 = 8"
+        assert expected in result.stdout.str()
+
+    def test_parametrization_keyerrors(self, testdir):
+        testdir.makepyfile("""
+            import pytest
+            @pytest.mark.parametrize("test_input, expected", [
+                ("3 + 5", 8),
+                ("2 + 4", 6),
+                ("6 * 9", 42),
+            ])
+            def test_math(test_input, expected):
+                '''Checking if {missing_key} = {expected}...'''
+                assert eval(test_input) == expected
+        """)
+
+        result = testdir.runpytest('--pspec')
+
+        assert result.ret == ExitCode.INTERNAL_ERROR
+        assert "INTERNALERROR> KeyError: 'missing_key'" in result.stdout.lines

